kolibri/plugins/learn/assets/src/modules/recommended/handlers.js

```diff
@@ -77,13 +78,39 @@ function _showLibrary(store, query, channels, baseurl) {
 }
 
 export function showLibrary(store, query, deviceId = null) {
-  if (deviceId) {
-    return setCurrentDevice(deviceId).then(device => {
-      const baseurl = device.base_url;
-      return fetchChannels({ baseurl }).then(channels => {
-        return _showLibrary(store, query, channels, baseurl);
-      });
+  /**
+   * ToDo: remove if block.
+   * Currently the channels & contentnode browser apis in studio
+   * are not able to load content using the the studio base url.
+   * Once studio is updated, this function will need to be refactored
+   * to use the else block code only.
+   *
+   * The if block is meant for UI viualization purposes only
+   * during development
+   */
+  if (deviceId === KolibriStudioId) {
+    RemoteChannelResource.getKolibriStudioStatus().then(({ data }) => {
+      if (data.status === 'online') {
+        RemoteChannelResource.fetchCollection().then(channels => {
+          //This is a hack to return kolibri channels.
+          store.commit('SET_ROOT_NODES', channels);
+
+          store.commit('CORE_SET_PAGE_LOADING', false);
+          store.commit('CORE_SET_ERROR', null);
+          store.commit('SET_PAGE_NAME', PageNames.LIBRARY);
+          return Promise.resolve();
+        });
+      }
     });
+  } else {
+    if (deviceId) {
+      return setCurrentDevice(deviceId).then(device => {
+        const baseurl = deviceId === KolibriStudioId ? plugin_data.studio_baseurl : device.base_url;
```

Comment:
maybe this is a naive question, but wouldn't it be possible to reuse the code from the device channel in the `showAvailableChannelsPage` function instead of rewritting something so similar?

Reply:
Ultimately, we should remove most of this special casing, once we have a version of Studio that we can test against that supports the v2 public content APIs, then all we need to do is pass in the Studio baseurl, and channels and resources should all be fetched properly from Studio.